pkg/osquery/table/kolide_target_membership.go

```diff
@@ -0,0 +1,123 @@
+package table
+
+import (
+	"context"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/kolide/launcher/pkg/osquery"
+	"github.com/kolide/launcher/pkg/pb/kt"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const targetMembershipKey = "target_membership"
+
+type TargetMembership struct {
+	logger       log.Logger
+	db           *bolt.DB
+	targetClient kt.KTargetClient
+}
+
+func NewTargetMembership(logger log.Logger, db *bolt.DB, targetClient kt.KTargetClient) TargetMembership {
+	return TargetMembership{
+		logger:       logger,
+		db:           db,
+		targetClient: targetClient,
+	}
+}
+
+func (tm *TargetMembership) Plugin() *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("id"),
+	}
+
+	return table.NewPlugin("kolide_target_membership", columns, tm.generate())
+}
+
+func (tm *TargetMembership) generate() table.GenerateFunc {
+	return func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+
+		targets, err := tm.GetTargetMembership(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting target membership")
+		}
+
+		results := make([]map[string]string, len(targets))
+		for i, t := range targets {
+			results[i] = map[string]string{"id": t.GetId()}
+		}
+
+		return results, nil
+	}
+}
+
+func (tm *TargetMembership) Run(ctx context.Context) error {
+	timeChan := time.NewTicker(30 * time.Second).C
+
+	for {
+		select {
+		case <-timeChan:
+			if err := tm.UpdateTargetMemberships(ctx); err != nil {
+				level.Debug(tm.logger).Log(
```

Comment:
Do we try to minimize log output in launcher? I would think this would be an info or error log, but just curios

Reply:
you're right - it looks like we log errors in the other cases. i'll update the level here as well